### ISSUE TITLE
Code style: inconsistent tab vs space indentation

### ISSUE THREAD

**Issue by alongalky** [iotaledger/iri#1413]
### Bug description
The IRI codebase contains inconsistent usage of tabs and spaces for indentation. This is annoying when contributing code, since code alignment will be editor-specific.

*Example (tabs):*
https://github.com/iotaledger/iri/blob/a28e954f5ac29dc3c4aefa5ca92fe9057e1b0bb6/src/main/java/com/iota/iri/service/dto/GetNodeInfoResponse.java#L168-L169

*Example (spaces):*
https://github.com/iotaledger/iri/blob/a28e954f5ac29dc3c4aefa5ca92fe9057e1b0bb6/src/main/java/com/iota/iri/service/dto/GetNodeInfoResponse.java#L33

*Examples (mix between the two in a single line):*
https://github.com/iotaledger/iri/blob/a28e954f5ac29dc3c4aefa5ca92fe9057e1b0bb6/src/main/java/com/iota/iri/service/dto/GetNodeInfoResponse.java#L153-L154

### Recommended fix
I suggest adding the appropriate styling rule to the automatic style tests.

**Comment by rajivshah3** [comment]
It might be helpful to add indentation rules to the `checkstyle.xml` and then enable the Checkstyle integration in Codacy so the bot will leave comments directly on the PR

**Comment by luca-moser** [comment]
I totally agree, it's very bothering having to check constantly whether I didn't accidentally format some random lines when I auto-format the file I am modifying. I expected everything to be already formatted accordingly to the style guide but there are so many issues:
* there are random lines where there are actually tabs on the line but no other content:
https://github.com/iotaledger/iri/blob/a28e954f5ac29dc3c4aefa5ca92fe9057e1b0bb6/src/main/java/com/iota/iri/service/tipselection/impl/WalkerAlpha.java#L81
* some lines exceed the line width defined by the style guide files
* the files from https://github.com/iotaledger/iri/blob/dev/STYLEGUIDE.md#formatting do not correctly format comments, or at least you will modify almost everything when you format most files (I am using IntelliJ with the Eclipse Code Formatter)

That's what I love so much about Go, everyone uses the same formatting and you know that when you auto-format your file, there's no discussions afterwards about changing random lines around. Perhaps we should finally extend the styleguide files to format everything like we want it to look like and then format the entire code base once.
